src/docbuild/utils/pidlock.py

```diff
@@ -5,165 +5,169 @@
 import logging
 import os
 import errno
-import os
 from pathlib import Path
-from typing import Self, Any
+from typing import TypeVar
+
+T = TypeVar("T", bound="PidFileLock")
 
 from ..constants import BASE_LOCK_DIR
 
+# --- Safe Logging Setup for CI / Pytest (prevents teardown ValueErrors) ---
+
+class _SafeStreamHandler(logging.StreamHandler):
+    """A logging handler that ignores 'I/O operation on closed file' errors during teardown."""
+    def emit(self, record):
+        try:
+            super().emit(record)
+        except ValueError as e:
+            if "I/O operation on closed file" in str(e):
+                # Ignore harmless teardown errors (common in Python 3.13 + pytest)
+                return
+            raise
+
+
```

Comment:
You have a similar class in `conftest.py`. Although the general idea is a good think, "poluting" the code with test related parts should be avoided. Can we integrate that in `conftest.py`?